data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java

```diff
@@ -260,52 +326,71 @@ public KustoOperationResult executeStreamingIngestFromBlob(String database, Stri
 
-            // We use UncloseableStream to prevent HttpClient From closing it
-            AbstractHttpEntity entity = isStreamSource ? new InputStreamEntity(new UncloseableStream(stream))
-                    : new StringEntity(new IngestionSourceStorage(blobUrl).toString(), ContentType.APPLICATION_JSON);
-            String response;
-            // trace executeStreamingIngest
-            response = MonitoredActivity.invoke(
-                    (SupplierTwoExceptions<String, DataServiceException, DataClientException>) () -> HttpPostUtils.post(httpClient, clusterEndpoint, entity,
-                            timeoutMs + CLIENT_SERVER_DELTA_IN_MILLISECS, headers),
-                    "ClientImpl.executeStreamingIngest");
-            return new KustoOperationResult(response, "v1");
-        } catch (KustoServiceQueryError e) {
-            throw new DataClientException(clusterEndpoint, "Error converting json response to KustoOperationResult:" + e.getMessage(), e);
-        } catch (KustoClientInvalidConnectionStringException | IOException e) {
-            throw new DataClientException(clusterUrl, e.getMessage(), e);
+        Map<String, String> headers = new HashMap<>();
+        String authorization = getAuthorizationHeaderValue();
+        String contentEncoding = null;
+        String contentType;
+        if (isStreamSource) {
+            contentEncoding = "gzip";
         }
-    }
 
-    private Long populateHeadersAndGetTimeout(ClientRequestProperties properties, Map<String, String> headers) throws DataClientException {
-        Long timeoutMs = null;
+        // This was a separate method but was moved into the body of this method because it performs a side effect
         if (properties != null) {
-            timeoutMs = determineTimeout(properties, CommandType.STREAMING_INGEST, clusterUrl);
             Iterator<Map.Entry<String, Object>> iterator = properties.getOptions();
             while (iterator.hasNext()) {
                 Map.Entry<String, Object> pair = iterator.next();
                 headers.put(pair.getKey(), pair.getValue().toString());
             }
         }
-        if (timeoutMs == null) {
-            timeoutMs = STREAMING_INGEST_TIMEOUT_IN_MILLISECS;
-        }
 
-        return timeoutMs;
+        try (InputStream ignored = (isStreamSource && !leaveOpen) ? stream : null) {
+            // Validate the endpoint
+            validateEndpoint();
+            BinaryData data;
+
+            if (isStreamSource) {
+                // We use UncloseableStream to prevent HttpClient From closing it
+                data = BinaryData.fromStream(new UncloseableStream(stream));
+                contentType = "application/octet-stream";
+            } else {
+                data = BinaryData.fromString(new IngestionSourceStorage(blobUrl).toString());
+                contentType = "application/json";
+            }
+
+            // Build the tracing object
+            HttpTracing tracing = HttpTracing
```

Comment:
I assume we can't unify this method with the other one better?

Reply:
I'm not happy with our current open-telemetry implementation, so I kept it decoupled to make it easy to remove later. I'd like to get rid of this class entirely. The azure-core library has its own built-in open-telemetry implementation that might be a better fit.